src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -1578,22 +1614,16 @@ bool ParserNumber::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     /// try to use more exact type: UInt64
 
-    char * pos_integer = buf;
-
-    errno = 0;
-    UInt64 uint_value = std::strtoull(buf, &pos_integer, 0);
-    if (pos_integer == pos_double && errno != ERANGE && (!negative || uint_value <= (1ULL << 63)))
+    if (parseNumber(buf, pos->size(), negative, 0, res))
```

Comment:
I can slowdown parsing of very long list of numbers, as they will be tried to be parsed as double first.